src/main/java/it/pagopa/interop/probing/eservice/operations/service/impl/EserviceServiceImpl.java

```diff
@@ -118,17 +106,54 @@ public void updateEserviceFrequency(UpdateEserviceFrequencyDto inputData)
         + " and frequency: " + eServiceToUpdate.pollingFrequency());
   }
 
+  @Override
   public SearchEserviceResponse searchEservices(Integer limit, Integer offset, String eserviceName,
-      String eserviceProducerName, Integer versionNumber, List<EserviceState> eServiceState) {
-    Page<EserviceView> eserviceList = eserviceViewRepository.findAll(
-        EserviceViewSpecs.searchSpecBuilder(eserviceName, eserviceProducerName, versionNumber,
-            eServiceState),
-        new OffsetLimitPageable(offset, limit,
-            Sort.by(ProjectConstants.ESERVICE_NAME_FIELD).ascending()));
-    return SearchEserviceResponse.builder()
-        .content(mapper.toSearchEserviceResponse(eserviceList.getContent()))
-        .offset(eserviceList.getNumber()).limit(eserviceList.getSize())
-        .totalElements(eserviceList.getTotalElements()).build();
+      String producerName, Integer versionNumber, List<EserviceStateFE> state) {
+
+    Page<EserviceView> eserviceList = null;
+    List<String> stateBE = state == null || state.isEmpty() ? new ArrayList<>()
+        : enumUtilities.convertListFromFEtoBE(state);
+
+    if (state == null || state.isEmpty() || (state.contains(EserviceStateFE.N_D)
```

Comment:
Use Objects to test null in each predicate

Reply:
Done